File: assignment-solutions/src/Practice/RemoveElements.java
package Practice;

// link : https://leetcode.com/problems/remove-element/

public class RemoveElements {
    public static void main(String[] args) {
        int[] arr = {0,1,2,2,3,0,4,2};
        int val = 2;
        System.out.println(removeElement(arr, val));
    }

    public static int removeElement(int[] arr, int val) {
        int k = 0;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] != val) {
                arr[k] = arr[i];
                k++;
            }
        }

        return k;
    }
}

File: assignment-solutions/src/Practice/Cpp/RemoveElements.cpp
#include <vector>
#include <iostream>
using namespace std;

// link:  https://leetcode.com/problems/remove-element/

class RemoveElements {
    public:
    int removeElements(vector<int>& arr, int val) {
        int k = 0;
        for (int i = 0; i < arr.size(); i++) {
            if (arr[i] != val) {
                arr[k] = arr[i];
                k++;
            }
        }
        return k;
    }
};

int main() {
    vector<int> arr = {0,1,2,2,3,0,4,2};
    int val = 2;
    RemoveElements rm;
    std::cout << rm.removeElements(arr, val);
}